commit bd3b0fbae33a9a4cc5e2daf049443d5cf03d4251
Author: Andreas Schwab <schwab@suse.de>
Date:   Mon Nov 5 12:47:30 2018 +0100

    libanl: properly cleanup if first helper thread creation failed (bug 22927)

diff --git a/resolv/gai_misc.c b/resolv/gai_misc.c
index e7c3b63cc5725b4f..80a2cff8353fcb6c 100644
--- a/resolv/gai_misc.c
+++ b/resolv/gai_misc.c
@@ -261,8 +261,11 @@ __gai_enqueue_request (struct gaicb *gaicbp)
 	      /* We cannot create a thread in the moment and there is
 		 also no thread running.  This is a problem.  `errno' is
 		 set to EAGAIN if this is only a temporary problem.  */
-	      assert (lastp->next == newp);
-	      lastp->next = NULL;
+	      assert (requests == newp || lastp->next == newp);
+	      if (lastp != NULL)
+		lastp->next = NULL;
+	      else
+		requests = NULL;
 	      requests_tail = lastp;
 
 	      newp->next = freelist;
